### PR TITLE
1.add R718n125 support

### PR DESCRIPTION
**Pull request by netvox-tech**
1.add R718n125 support

### CODE REVIEW

File: decoders/connector/netvox/r718n125/v1.0.0/payload.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+function Decoder_r718n125(input, port) {
```
Comment (mateuscardosodeveloper) [inline decoders/connector/netvox/r718n125/v1.0.0/payload.ts:1]: Add type annotation for the function, example:
```ts
function Decoder_r718n125(input: Buffer, port: number) {
```

File: decoders/connector/netvox/r718n125/v1.0.0/payload.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    case 6: // 0x06
+      return decodePort6_r718n125(input);
+    case 7: // 0x07
+      return decodePort7_r718n125(input);
+    default:
+      throw new Error("no port configured");
+  }
+}
+
+function decodePort6_r718n125(bytes) {
```
Comment (mateuscardosodeveloper) [inline decoders/connector/netvox/r718n125/v1.0.0/payload.ts:12]: Add type annotation for the function.

File: decoders/connector/netvox/r718n125/v1.0.0/payload.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        unit: "mA",
+      });
+      data.push({variable: "multiplier", value: Number(bytes[6])});
+      data.push({variable: "lowCurrent", value: (bytes[7] & 1) > 0 ? 'alarm' : 'noalarm'});
+      data.push({variable: "highCurrent", value: (bytes[7] >> 1 & 1) > 0 ? 'alarm' : 'noalarm'});
+      return data;
+  }
+}
+
+function decodePort7_r718n125(bytes) {
```
Comment (mateuscardosodeveloper) [inline decoders/connector/netvox/r718n125/v1.0.0/payload.ts:39]: Add type annotation for the function

File: decoders/connector/netvox/r718n125/v1.0.0/payload.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+const payload_r718n125 = payload.find((x) => x.variable === "payload_raw" || x.variable === "payload" || x.variable === "data");
+const port_r718n125 = payload.find((x) => x.variable === "port" || x.variable === "fPort" || x.variable === "fport")?.value;
+
+if (payload_r718n125) {
+  try {
+    const buffer = Buffer.from(payload_r718n125.value, "hex");
+    const data = Decoder_r718n125(buffer, port_r718n125);
+    payload = payload.concat(data.map((x) => ({...x, serie: payload_r718n125.group, time: payload_r718n125.time})));
```
Comment (mateuscardosodeveloper) [inline decoders/connector/netvox/r718n125/v1.0.0/payload.ts:80]: Field "serie" is deprecated, replace to "group":
```ts
payload = payload.concat(data.map((x) => ({ ...x, group: payload_r718n125.group, time: payload_r718n125.time })));
```

File: decoders/connector/netvox/r718n125/v1.0.0/payload.ts (hunk 1 above)
Comment (mateuscardosodeveloper) [inline decoders/connector/netvox/r718n125/v1.0.0/payload.ts:1]: Install the package `npm install -D @tago-io/sdk` and declare the variable payload.
```ts
import { DataToSend } from "@tago-io/sdk/lib/types";

declare let payload: DataToSend[];
```

Probably when you declare the payload variable you will get errors on line 82 and 83 replace it with:
```ts
    const buffer = Buffer.from(payload_r718n125?.value as string, "hex");
    const data = Decoder_r718n125(buffer, port_r718n125 as number);
```

File: decoders/connector/netvox/r718n125/v1.0.0/payload.test.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
+/* eslint-disable unicorn/numeric-separators-style */
+import {describe, test, expect} from "vitest";
+import {decoderRun} from "../../../../../src/functions/decoder-run";
+
+const file_path = "decoders/connector/netvox/r718n125/v1.0.0/payload.ts" as const;
+
+function preparePayload(payloadHex, port) {
```
Comment (mateuscardosodeveloper) [inline decoders/connector/netvox/r718n125/v1.0.0/payload.test.ts:7]: Add type annotation for the function.

**Review by mateuscardosodeveloper — CHANGES_REQUESTED** [review]

**Comment by Freddyminu** [comment]
Hi @netvox-tech ,

Thank you for creating this Pull Request! I have made some modifications and created a new PR, which has been merged. Therefore, this PR is no longer needed.

**Pull request closed without merging**